Update microarray analyses: differential expression

cansavvy opened the pull request:
## Purpose:
Closes #182 

## Strategy
Beside making the analysis fit the style guide (#173), I also: 

- I added a bit where we change an annoying column name (`genotype/variation`) to a less annoying name. Do I spend too much time on explaining this? 
- Removed the Q-Q plot entirely. I feel like most people like volcano plots better anyway (tell me if we want this back though). 
- Switched the limma volcano plot to the snazzier `EnhancedVolcano` plot
- Added a "sanity check" plot for a top gene. I'm concerned it takes a bit too much work to get the dataset there, but I do like the sanity check plot itself. Let me know what you think

## Concerns/Questions for reviewers:
- How do you feel about the order of the steps? 
- Are there places that should be trimmed down or cut out? 

## Analysis Pull Request Check List (roughly in order):

### Content checks
* [x] All `{{BLANKS}}` have been replaced with the correct content.
* [x] [Sources are cited](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#citing-sources-in-text)
* [x] Seed is set (if applicable)

### Formatting Checks
* [x] Removed any manual numbering of sections.
* [x] Removed any instances of chunk naming.
* [x] Spell checked any Rmd file or md file.
* [x] Comments and documentation are up to date.

### Add datasets to S3
* [x] Added [data and metadata files to S3](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#adding-datasets-to-the-S3-bucket).

### Docker/Snakemake rendering components
* [x] Added the `.html` [link to the navigation bar](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#add-new-analyses-to-the-navbar).
* [x] Any not yet added packages needed for this analysis have been added to the Dockerfile and it successfully builds.
* [x] In the Docker container, [snakemake was run for rendering](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#how-to-re-render-the-notebooks).

cbethell commented on 02-microarray/differential-expression_microarray_01.Rmd:
```diff
@@ -0,0 +1,349 @@
+
+By default, results are ordered by largest `B` (the log odds value) to the smallest, which means your most differentially expressed genes should be toward the top. 
+
+See the help page by using `?topTable` for more information and options for this table. 
```
```suggestion
See the help page by using `?topTable` for more information on arguments like `adjust.method` and options for this table.
```

This is not completely necessary but I thought it would be helpful to explicitly include a note that more info on the `adjust.method` argument that is mentioned on line 240 (but not actually used here) can also be found by using `topTable`.

cansavvy replied:
Hmm... I don't feel like we need to rehash about `adjust.method` as an argument. This is enough to get them to the help page and they can find out about that from there. But I will add a reminder about the help page after I tell them about `adjust.method` the first time.

cbethell reviewed (commented):
This looks great @cansavvy! Just have a few comments/suggestions below.

cbethell commented on 02-microarray/differential-expression_microarray_01.Rmd:
````diff
@@ -0,0 +1,349 @@
+  y = "adj.P.Val"
+)
+```
+
````
We should probably add a sentence here commenting on the possible interpretation of the plot above as well.

cansavvy replied:
I've added some more guidance about `EnhancedVolcano` plot! Good call.

cansavvy commented on 02-microarray/differential-expression_microarray_01.Rmd:
```diff
@@ -0,0 +1,349 @@
+## About the dataset we are using for this example
+
+For this example analysis, we will use this [creb overexpression zebrafish](https://www.refine.bio/experiments/GSE71270/creb-overexpression-induces-leukemia-in-zebrafish-by-blocking-myeloid-differentiation-process).
+@Trenago2016 measured microarray gene expression of zebrafish samples overexpressing human CREB, as well as control samples.
```
I actually don't want brackets since this is an [in-text citation](https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#Citations), but the reason its not showing up right is I spelled it wrong. Should be `@Tregnago2016`.

cbethell replied:
Ahh, gotcha 👍

cansavvy commented on 02-microarray/differential-expression_microarray_01.Rmd:
```diff
@@ -0,0 +1,349 @@
+    metadata,
+    refinebio_accession_code,
+    genotype
+  ))
```
Odd. The notebook ran fine for me. I'll look into that.

cbethell replied:
Ah that was indeed the case (not on the most up to date Docker image), sorry for the confusion!

cansavvy replied:
No problemo!

cansavvy commented on 02-microarray/differential-expression_microarray_01.Rmd:
````diff
@@ -0,0 +1,349 @@
+    refinebio_accession_code,
+    genotype
+  ))
+```
````
I'll show a preview of `top_gene_df`. That sounds good to me! I agree, I don't think it hurts to leave this in.

cansavvy commented:
@cbethell I believe I addressed your changes, but I also added more guidance about EnhancedVolcano plot in general (and I carried this over to the rna-seq differential expression module as well). See what you think about the the edits I made to the `volcano plot` section.

cbethell reviewed (approved):
Looks great! Just have a couple minor formatting suggestions below, but otherwise LGTM!

cansavvy commented on 02-microarray/differential-expression_microarray_01.Rmd:
````diff
@@ -0,0 +1,371 @@
+```
+
+These results make sense.
+The overexpressing CREB group samples have much higher expression values for ENSDARG00000104315 than the control samples do. 
````
I thought about this, but left off the backticks. My reasoning is that the gene exists outside of our code and this sentence we are referring more to the gene as a concept than specifically referring to it as a column name. What do you think?

cansavvy commented on 02-microarray/differential-expression_microarray_01.Rmd:
````diff
@@ -0,0 +1,371 @@
+top_gene_df
+```
+
+Now let's plot the data for ENSDARG00000104315 using our `top_gene_df`.
````
I think according to my reasoning below, I *should* bring the backticks here, if you agree.

cbethell replied:
I do agree, with both you reasoning and the idea that the backticks should be brought back here in this instance, but not in the instance below 👍

cansavvy replied:
Will do.

cansavvy replied:
I added one line in the CONTRIBUTING.md that explains this idea.

cansavvy commented:
I've pushed a few minor edits since you gave me that approval that mainly have to do with rendering. Can you check over that the commits I've done since your approval and see if they seem ship shape to you? 

The main thing I had to fix was some how the UMAP from RNA-seq got dropped (or never got there in the first place?) from the Snakefile and the navbar.

cansavvy commented:
In master, UMAP is not in the Snakefile: https://github.com/AlexsLemonade/refinebio-examples/blob/24cfaae5747f00de811ecf52fb51cf7e612ee950/Snakefile#L11

But in the original PR it was there: https://github.com/AlexsLemonade/refinebio-examples/pull/193/files#diff-b0ca7e384af213e2f691bf0435d08557

Some merge and conflict resolve must have lost it?

cbethell commented:
> I've pushed a few minor edits since you gave me that approval that mainly have to do with rendering. @cbethell Can you check over that the commits I've done since your approval and see if they seem ship shape to you?
> 
> The main thing I had to fix was some how the UMAP from RNA-seq got dropped (or never got there in the first place?) from the Snakefile and the navbar.

Took another look and everything looks good to me 👍

> Some merge and conflict resolve must have lost it?

Strange that we lost rnaseq UMAP in the Snakefile, I agree with your statement above (the best case I could think of), but it's fixed now!
